dvc/exceptions.py

```diff
@@ -268,8 +268,8 @@ def __init__(self, target_infos, stats=None):
         self.stats = stats
         targets = [str(t) for t in target_infos]
         m = (
-            "Checkout failed for following targets:\n{}\nDid you "
-            "forget to fetch?".format("\n".join(targets))
+            "Checkout failed for following targets:\n{}\nIs your "
+            "cache up to date?".format("\n".join(targets))
```

Comment:
Hm, not sure if this makes it clearer. The OP https://github.com/iterative/dvc/issues/4316 was mentioning that he forgets to push stuff on another machine, and this still doesn't seem to be directing in the right direction. Maybe we should actually add a link with common issues for such warning and have an article about it on dvc.org? Kinda like we do with https://dvc.org/doc/user-guide/troubleshooting (maybe that's even a good place for such hints as well). What do you think? CC @braaannigan

Reply:
The cache is a bit mysterious for new users and this message wouldn't necessarily cause me to think about updating the cache from my other machine. I like the idea about having a link to a web page with common issues

Reply:
I am not sure changing the message will do the trick here. I wrote some thoughts on that under the [original issue](https://github.com/iterative/dvc/issues/4316#issuecomment-669138615).